sql: TestScanBatches failed

[(sql).TestScanBatches failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2174289&tab=buildLog) on [master@44c362600dbed9d58be8daf64558543745698736](https://github.com/cockroachdb/cockroach/commits/44c362600dbed9d58be8daf64558543745698736):

```

Goroutine 555 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:174 +0x595
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:446 +0x7ee
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:241 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:216 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:271 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:255 +0x2b6
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).Update.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:284 +0x2b7
  github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:707 +0x5e
  github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:811 +0xe9
  github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:706 +0x11c
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).Update()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:283 +0x126
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance.(*Instance).extendSession()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance/slinstance.go:161 +0x3b0
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance.(*Instance).heartbeatLoop()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance/slinstance.go:207 +0x2e4
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance.(*Instance).heartbeatLoop-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance/slinstance.go:183 +0x55
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:222 +0x1b8

Goroutine 1311 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1202 +0xa6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1200 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1117 +0x2ff
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:32 +0x15b
  main.main()
      _testmain.go:660 +0x223
==================
FAIL	github.com/cockroachdb/cockroach/pkg/sql	112.023s

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestScanBatches PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestScanBatches.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestScanBatches failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2177510&tab=buildLog) on [master@457f1af5d19e0bcef9b5ea9e9125b3332717a86c](https://github.com/cockroachdb/cockroach/commits/457f1af5d19e0bcef9b5ea9e9125b3332717a86c):

```
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:174 +0x595
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:446 +0x7ee
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:241 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:216 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:271 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:255 +0x2b6
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteExpiredSessions.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:226 +0x216
  github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:707 +0x5e
  github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:811 +0xe9
  github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:706 +0x11c
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteExpiredSessions()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:225 +0xfc
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteSessionsLoop()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:218 +0x101
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).deleteSessionsLoop-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:203 +0x55
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:222 +0x1b8

Goroutine 1040 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1202 +0xa6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1200 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1117 +0x2ff
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:32 +0x15b
  main.main()
      _testmain.go:660 +0x223
==================
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestScanBatches632231892
--- FAIL: TestScanBatches (3.52s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestScanBatches632231892
    test_log_scope.go:63: use -show-logs to present logs inline
    testing.go:853: race detected during execution of test

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestScanBatches PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestScanBatches.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestScanBatches failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2191440&tab=buildLog) on [master@22a31b641abbb5ec59023eb73dde63146159b515](https://github.com/cockroachdb/cockroach/commits/22a31b641abbb5ec59023eb73dde63146159b515):

```
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:446 +0x7ee
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:241 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:216 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:271 +0x14f
  github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:255 +0x2b6
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).Update.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:286 +0x2b7
  github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:707 +0x5e
  github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:811 +0xe9
  github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:706 +0x11c
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage.(*Storage).Update()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slstorage/slstorage.go:285 +0x126
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance.(*Instance).extendSession()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance/slinstance.go:161 +0x3b0
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance.(*Instance).heartbeatLoop()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance/slinstance.go:207 +0x2e4
  github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance.(*Instance).heartbeatLoop-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance/slinstance.go:183 +0x55
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:222 +0x1b8

Goroutine 359 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1202 +0xa6
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1200 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1117 +0x2ff
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:32 +0x15b
  main.main()
      _testmain.go:672 +0x223
==================
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestScanBatches809398458
--- FAIL: TestScanBatches (3.49s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestScanBatches809398458
    test_log_scope.go:63: use -show-logs to present logs inline
    testing.go:853: race detected during execution of test

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestScanBatches PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestScanBatches.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/scan_test.go
+++ b/pkg/sql/scan_test.go
@@ -132,6 +132,9 @@ func TestScanBatches(t *testing.T) {
 	s, db, _ := serverutils.StartServer(
 		t, base.TestServerArgs{UseDatabase: "test"})
 	defer s.Stopper().Stop(context.Background())
+	if _, err := db.Exec("SET CLUSTER SETTING server.sqlliveness.heartbeat='1000s'"); err != nil {
+		t.Fatal(err)
+	}
 
 	if _, err := db.Exec(`CREATE DATABASE IF NOT EXISTS test`); err != nil {
 		t.Fatal(err)

